### DISCUSSION TITLE
Question regarding `ConfigureAwait(false)`

### DISCUSSION THREAD

**Discussion by MartyIX** [brminnick/HackerNews#38] [Q&A] [answered]
Hi,

I have been trying to figure out a question of mine regarding:

https://github.com/brminnick/HackerNews/blob/aaad2453ea329005a5e50b1ec5cdaba350b8dc7b/src/HackerNews/ViewModels/NewsViewModel.cs#L76-L106

Basically, my question is: Is it OK to use `ConfigureAwait(false)` here: https://github.com/brminnick/HackerNews/blob/aaad2453ea329005a5e50b1ec5cdaba350b8dc7b/src/HackerNews/ViewModels/NewsViewModel.cs#L78

?

The reason I'm asking is that `IsListRefreshing` is not necessarily set on the UI thread:

https://github.com/brminnick/HackerNews/blob/aaad2453ea329005a5e50b1ec5cdaba350b8dc7b/src/HackerNews/ViewModels/NewsViewModel.cs#L104

And I wonder whether one can set view model properties (used in data bindings) on non-UI threads. Is it or is it not?

Best regards!

**Comment by TheCodeTraveler** [comment, marked as answer]
Good question! 

Short answer: Yes, in .NET MAUI it's safe to update properties in the ViewModel that are bound to the view from any thread.

.NET MAUI (and Xamarin.Forms) marshall binding updates to the Main (UI) Thread, so we're totally safe setting a property in our ViewModel from any thread because MAUI's binding engine will bring the updates to the UI Thread for us.

You'll want to stay aware around this, though. There has been talks amongst the engineering team to no longer marshall every binding to the UI Thread, and I've seen some instances in newer versions of .NET MAUI where it wasn't properly marshaling CollectionView updates to the UI Thread. It'd be a substantial breaking change, and I don't they'd break all of our code, but it could happen!

**Reply by MartyIX** [reply]
Thank you for the answer. Much appreaciated!

**Reply by MartyIX** [reply]
> .NET MAUI (and Xamarin.Forms) marshall binding updates to the Main (UI) Thread, so we're totally safe setting a property in our ViewModel from any thread because MAUI's binding engine will bring the updates to the UI Thread for us.

Do you happen to know where in MAUI's source code I could find this marshalling?

**Reply by TheCodeTraveler** [reply]
In [`BindingExpression.OnPropertyChanged()`](https://github.com/dotnet/maui/blob/1b86f6583ed97c82de67a8c29ee0ad7193b45059/src/Controls/src/Core/BindingExpression.cs#L750), the binding expression is marshaled to the UI Thread via `Dispatcher.DispatchIfRequired()`:
https://github.com/dotnet/maui/blob/1b86f6583ed97c82de67a8c29ee0ad7193b45059/src/Controls/src/Core/BindingExpression.cs#L750
